tests/js/common/shell/shell-figures-rocksdb.js

```diff
@@ -0,0 +1,495 @@
+        assertEqual("primary", indexes[0].type);
+        assertEqual(0, indexes[0].id);
+        assertEqual(100, indexes[0].count);
+        assertEqual("rocksdb-ttl", indexes[1].type);
+        assertEqual(idxs[0].id, cn + '/' + indexes[1].id);
+        assertEqual(0, indexes[1].count);
+      } finally {
+        db._drop(cn);
+      }
+    },
+    
+    testDetailedTtl: function () {
+      let c = db._create(cn, { numberOfShards: 4 });
+      try {
+        let idxs = [];
+        idxs.push(c.ensureIndex({ type: "ttl", fields: ["value1"], expireAfter: 100000 }));
+        let dt = (new Date()).toISOString();
+        for (let i = 0; i < 100; ++i) {
+          c.insert({ value1: dt, value2: "test" + i });
+        }
+        let figures = c.figures(true).engine;
+        assertEqual(100, figures.documents);
+
+        let indexes = figures.indexes;
+        assertEqual(2, indexes.length);
+        assertEqual("primary", indexes[0].type);
+        assertEqual(0, indexes[0].id);
+        assertEqual(100, indexes[0].count);
+        assertEqual("rocksdb-ttl", indexes[1].type);
+        assertEqual(idxs[0].id, cn + '/' + indexes[1].id);
+        assertEqual(100, indexes[1].count);
+      } finally {
+        db._drop(cn);
+      }
+    },
+    
+    testDetailedGeoInvalid: function () {
+      let c = db._create(cn, { numberOfShards: 4 });
+      try {
+        let idxs = [];
+        idxs.push(c.ensureIndex({ type: "geo", fields: ["lat", "lon"] }));
+        for (let i = 0; i < 100; ++i) {
+          c.insert({ lat: "piff", lon: "paff" });
+        }
+        let figures = c.figures(true).engine;
+        assertEqual(100, figures.documents);
+
+        let indexes = figures.indexes;
+        assertEqual(2, indexes.length);
+        assertEqual("primary", indexes[0].type);
+        assertEqual(0, indexes[0].id);
+        assertEqual(100, indexes[0].count);
+        assertEqual("geo", indexes[1].type);
+        assertEqual(idxs[0].id, cn + '/' + indexes[1].id);
+        assertEqual(0, indexes[1].count);
+      } finally {
+        db._drop(cn);
+      }
+    },
+    
+    testDetailedGeo: function () {
+      let c = db._create(cn, { numberOfShards: 4 });
+      try {
+        let idxs = [];
+        idxs.push(c.ensureIndex({ type: "geo", fields: ["lat", "lon"] }));
+        for (let i = 0; i < 100; ++i) {
+          c.insert({ lat: 50 - i, lon: 50 - i });
+        }
+        let figures = c.figures(true).engine;
+        assertEqual(100, figures.documents);
+
+        let indexes = figures.indexes;
+        assertEqual(2, indexes.length);
+        assertEqual("primary", indexes[0].type);
+        assertEqual(0, indexes[0].id);
+        assertEqual(100, indexes[0].count);
+        assertEqual("geo", indexes[1].type);
+        assertEqual(idxs[0].id, cn + '/' + indexes[1].id);
+        assertEqual(100, indexes[1].count);
+      } finally {
+        db._drop(cn);
+      }
+    },
+    
+    testDetailedSmartEdge: function () {
```

Comment:
I would think you would need to separate out this kind of thing into enterprise-only tests, but maybe I'm wrong?

Reply:
This is already guarded with the if condition
```
      if (!isCluster || !isEnterprise) {
        return;
      }
```
so that it will not be executed in a non-cluster or non-enterprise environment.
Of course, I could have moved the one test to the enterprise directory, so I would have had a nice split into community and enterprise tests. But logically all the tests in this file belong together, so I preferred to keep them in one file and add if guards here.